lib/Common/CommonDefines.h

```diff
@@ -168,6 +168,7 @@
 #define SYSINFO_IMAGE_BASE_AVAILABLE 1
 #define ENABLE_CONCURRENT_GC 1
 #define ENABLE_ALLOCATIONS_DURING_CONCURRENT_SWEEP 1 // Only takes effect when ENABLE_CONCURRENT_GC is enabled.
+#define ALLOCATIONS_DURING_CONCURRENT_SWEEP_HEAPBLOCK_THRESHOLD 5000 // Number of blocks a heap bucket needs to have before allocations during concurrent sweep feature kicks-in.
```

Comment:
Why is this defined here and not in RecyclerHeuristic.h?

Since you're touching this code, can you refactor the defines slightly? You can probably move ENABLE_CONCURRENT_GC outside of the _WIN32 define, and just check for #if ENABLE_VALGRIND?

Also, why is allocation during concurrent sweep disabled on xplat? This feature would benefit node scenarios...

Reply:
I guess that's because we don't have SLIST support on linux...

---
In reply to: [148163317](https://github.com/Microsoft/ChakraCore/pull/4104#discussion_r148163317) [](ancestors = 148163317)

Reply:
Good point, moved the heuristic to RecyclerHeuristic.h. Lei is right about the SLIST support. We will look at adding this support in future. I will try to do the refactoring if I can now.

---
In reply to: [148163317](https://github.com/Microsoft/ChakraCore/pull/4104#discussion_r148163317) [](ancestors = 148163317)

Reply:
Just FYI; locks may eventually involve for xplat SLIST. There are platform specific weirdness with interlocked exchange utilities that we may not depend on our favorite arch.

Last time I remember the potential gain from having the SLIST wasn’t big hence I had reduced the priority significantly. Is this something we may reconsider SLIST xplat priority?